scripts/poChecker.py

```diff
@@ -104,33 +106,47 @@ def _checkMessages(self):
 			if line.startswith(self.FUZZY):
 				nextFuzzy = True
 				continue
-			elif line.startswith(self.MSGID):
+			elif line.startswith(self.MSGID) and not line.startswith(self.MSGID_PLURAL):
 				# New message.
-				self._msgstr = self._finishString()
-				if self._msgstr:
+				if self._msgstr is not None:
+					self._msgstr[-1] = self._finishString()
 					# Check the message we just handled.
 					self._checkMessage()
 				command = self.MSGID
 				start = command
 				self._messageLineNum = lineNum
 				self._fuzzy = nextFuzzy
 				nextFuzzy = False
-			elif line.startswith(self.MSGSTR):
+			elif line.startswith(self.MSGID_PLURAL):
 				self._msgid = self._finishString()
+				command = self.MSGID_PLURAL
+				start = command				
+			elif line.startswith(self.MSGSTR):
+				if command == self.MSGID:
+					self._msgid = self._finishString()
+				elif command == self.MSGID_PLURAL:
+					self._msgid_plural = self._finishString()
+				elif command == self.MSGSTR:
+					self._msgstr[-1] = self._finishString()
+					self._checkMessage()
+				else:
+					raise RuntimeError(f'Unexpected command before line {lineNum}: {command}')
+				if command != self.MSGSTR:
+					self._msgstr = []
+				self._msgstr.append("")
 				command = self.MSGSTR
-				start = command
+				start = line[:line.find(' ')]
```

Comment:
This function is quite complex, and would probably through the relevant lint error. Could this new block be reformatted into a helper function (e.g. `_handleMsgStr`)